apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useCoinMetadata } from '@mysten/core';
+import { Sui, Unstaked } from '@mysten/icons';
+import { type CoinMetadata } from '@mysten/sui.js';
+import clsx from 'classnames';
+
+import { Text } from '../../text';
+
+function CoinIcon({ coinMetadata }: { coinMetadata?: CoinMetadata | null }) {
+    if (coinMetadata?.symbol === 'SUI') {
```

Comment:
Only use the fully qualified type to determine if we special case this. Symbols are not the right way to do it. If this is how the explorer works, please change this there too.